protocol/x/clob/keeper/clob_pair.go

```diff
@@ -553,3 +553,26 @@ func (k Keeper) IsPerpetualClobPairActive(
 
 	return clobPair.Status == types.ClobPair_STATUS_ACTIVE, nil
 }
+
+// UnsafeSetClobPair sets a specific `ClobPair` in the store from its index.
+func (k Keeper) UnsafeSetClobPair(ctx sdk.Context, clobPair types.ClobPair) {
+	b := k.cdc.MustMarshal(&clobPair)
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.ClobPairKeyPrefix))
+	// Write the `ClobPair` to state.
+	store.Set(types.ClobPairKey(clobPair.GetClobPairId()), b)
+
+	// Send UpdateClobPair to indexer.
+	k.GetIndexerEventManager().AddTxnEvent(
```

Comment:
Added TxnEvent for `UpdateClobPair`, @Christopher-Li PTAL

Reply:
This is not actually related to this issue, but this indexer event should also be emitted during `keeper.UpdateClobPair` as well I believe. Think it was something we missed adding when working on the add new markets project (which relied on the new MsgUpdateClobPair) message

Reply:
This looks good, I'm assuming that this should be one of the last events in the block? After all subaccountUpdateEvents and OrderFillEvents?

Reply:
Yes, this is currently called in the upgrade handler which should execute in the `EndBlock` ([reference](https://docs.cosmos.network/main/modules/upgrade))

Reply:
Non-blocking nit: Since this specific function is currently only run in the `EndBlocker`, we could change to using the correct event type.
e.g.
```
indexerEventManager.AddBlockEvent(
	ctx,
	indexerevents.SubtypeUpdateClobPair,
	indexer_manager.GetB64EncodedEventMessage(
		indexerevents.NewUpdateClobPairEvent(
			clobPair.Id,
			clobPair.Status,
			clobPair.QuantumConversionExponent,
			clobPair.SubticksPerTick,
			clobPair.StepBaseQuantums,
		),
	),
	indexer_manager.IndexerTendermintEvent_BLOCK_EVENT_END_BLOCK,
)
```